File: MinjiJu/2023-01-17-Encode-and-Decode-Strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        for (string &s : strs) encoded += to_string(s.size()) + '.' + s;
	    return encoded;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        for (int i = 0, n = 0; i < s.size(); i += n, n = 0) {
		    while (isdigit(s[i])) n = 10 * n + (s[i++] - '0');
		    decoded.push_back(s.substr(++i, n));
	    }
	    return decoded;
    }
private:
    string encoded = "";
    vector<string> decoded = {};
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: Heeseop An/2023-01-17-Encode-And-Decode-Strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        encoded = ""
        for s in strs:
            encoded += str(len(s)) + "#" + s

        return encoded


    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """

        decoded = []
        i = 0
        while i < len(s):
            j = i
            while s[j] != "#":
                j += 1

            length = int(s[i:j])
            decoded.append(s[j + 1: j + 1 + length])
            i = j + 1 + length

        return decoded






# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))